thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/task/TaskDriver.java

```diff
@@ -183,8 +192,11 @@ private TaskDTO acquireTask() {
       try {
         // randomize fetching head and tail to reduce synchronized patterns across threads (and hosts)
         boolean orderAscending = System.currentTimeMillis() % 2 == 0;
+
+        // find by task type to separate online task from a normal task
+        TaskType type = this.isOnline ? TaskType.DETECTION_ONLINE : TaskType.DETECTION;
         anomalyTasks = taskDAO
-            .findByStatusOrderByCreateTime(TaskStatus.WAITING, driverConfiguration.getTaskFetchSizeCap(),
+            .findByStatusAndTypeOrderByCreateTime(TaskStatus.WAITING, type, driverConfiguration.getTaskFetchSizeCap(),
```

Comment:
@jasonyanwenl Sorry we missed this one earlier. Will this making it only picking up the Detection task here? This could be a serious issue because it won't be picking up other tasks, for example, sending out the notification, run detection onboarding, etc. Could you double-check?

@vincentchenjl @akshayrai we may need to hold on deployment to prod. 
@suvodeep-pyne thanks for reporting this

Reply:
Thank you so much. Really sorry about this. This is an issue. I will send a PR soon today.